content/matplotlib/concepts/pyplot/terms/pie/pie.md

```diff
@@ -0,0 +1,119 @@
+---
+Title: '.pie()'
+Description:  'Creates a pie chart of array x.'
+Subjects:
+  - 'Data Science'
+  - 'Data Visualization'
+Tags:
+  - 'Graphs'
+  - 'Libraries'
+  - 'Matplotlib'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/data-science'
+---
+
+The **`.pie()`** method in the matplotlib library is used to draw a pie chart, showing the percentage of the parts with respect to the whole data.
+
+## Syntax
+
+```pseudo
+matplotlib.pyplot.pie(x, explode, labels, colors, autopct, pctdistance, shadow, labeldistance, startangle, radius, counterclock, wedgeprops, textprops, center, frame, rotatelabels, normalize, hatch, data)
+```
+
+'x' parameter is required, and represents a 1D array-like object. Other parameters are optional and modify plot features like wedge colors and/or labels.
+
+`.pie()` takes the following arguments:
+
+-  `x`: 1D array-like object that specifies the wedge sizes.
+
+-  `explode`: Array-like object which specifies the fraction of the radius with which to offset each wedge.
+
+-  `labels`: List of strings providing the labels for each wedge.
+
+-  `colors`: Array-like sequence of colors through which the pie chart will cycle.
+
+-  `autopct`: A string or function used to label the wedges with their numeric value. The label will be placed inside the wedge.
+
+-  `pctdistance`: Float that sets the relative distance along the radius at which the the text generated by _autopct_ is drawn.
+
+-  `shadow`: Boolean value, draws a shadow beneath the pie.
+
+-  `labeldistance`: Float that sets the relative distance along the radius at which the labels are drawn.
+
+-  `startangle`: Float that sets the angle by which the start of the pie is rotated, counterclockwise from the x-axis.
+
+-  `radius`: Float that sets the radius of the pie.
+
+-  `counterclock`: Boolean to specify fractions direction, clockwise or counterclockwise.
+
+-  `wedgeprops`: Dict of arguments passed to each `patches.Wedge`
+
+-  `textprops`: Dict of arguments to pass to the text objects.
+
+-  `center`: (float, float), the coordinates of the center of the chart.
+
+-  `frame`: Bool, plot Axes frame with the chart if true.
+
+-  `rotatelabels`: Rotate each label to the angle of the corresponding slice if true.
+
+-  `normalize`: Bool, When _True_, always make a full pie by normalizing x. _False_makes a partial pie.
+
+-  `hatch`: Str or list, Hatching pattern applied to all pie wedges or sequence of patterns through which the chart will cycle.
+
+  ## Examples
+
+ 
+Examples below demonstrate the use of `.pie()` to plot values and show the functionality of some arguments.
+
+A simple pie chart: 
+
+```py
+
+# Import libraries
+from matplotlib import pyplot as plt
+
+# Creating dataset
+pizza = ['CHEESE','PEPPERONI','VEGGIE','MEAT','BBQ CHICKEN','HAWAIIAN']
+data = [23,  17,  35,  29,  12,  41]
```

Comment:
these values add up to greater than 100

Reply:
They shouldn't have to.

Reply:
Respectfully, I must insist this is incorrect. If you have a poll in which you ask everyone for a single "favorite" topping you will always get values that add up to 100 (e.g. 40% pepperoni , 30% sausage, 25% mushroom, 5% cheese). If you have data that does not add up to 100, it should not be displayed with a pie plot. Using a pie plot in circumstances in which the data does not add up to 100 is a visualization no-no.

Reply:
What if you're counting the number of pizzas sold and you want to have the pie chart to display the percentage of each? You're asking the developer to do the math that `.pie()` does automatically for you.

Reply:
Technically, this first example is okay. However, the second example uses the following data:
popularity = [22.2,  17.6,  8.8,  8,  7.7,  6.7]
The name of the variable and the floating point values suggest percentages. But, these values do not add up to 100. If you have vote counts representing the popularity of toppings there should only be whole numbers. It's impossible to have .2 of a vote. Therefore, the data doesn't make sense in this context.

Reply:
I've updated the second example to use whole numbers.